scripts/shell/bt_nus_shell.py

```diff
@@ -1,84 +1,88 @@
+
+
+CFG_TAG = 1
 
 parser = argparse.ArgumentParser(description='BLE serial daemon.')
 parser.add_argument('--com', help='COM port name (e.g. COM110).',
-                    nargs='*', required=False, default="")
+                    nargs='*', required=True, default="")
 parser.add_argument('--snr', help='segger id.',
                     nargs=1, required=False, default="")
 parser.add_argument('--family', help='chip family.',
                     nargs=1, required=False, default="")
 parser.add_argument('--name', help='Device name (advertising name).',
-                    nargs='*', required=True, default=None)
-args = parser.parse_args()
-
-def ble_driver_init(conn_ic_id):
-    global BLEDriver, Flasher, DfuTransportBle
-    from pc_ble_driver_py import config
-    config.__conn_ic_id__ = conn_ic_id
+                    nargs='*', required=False, default=None)
+parser.add_argument('--port', help='Local port to use.',
+                    required=False, default=8889)
 
-ble_driver_init(args.family[0])
+args = parser.parse_args()
 
-from nordicsemi.dfu.dfu_transport_ble import DFUAdapter
-from pc_ble_driver_py.exceptions      import NordicSemiException, IllegalStateException
-from pc_ble_driver_py.ble_driver      import config,Flasher,BLEDriver, BLEDriverObserver, BLEEnableParams, BLEUUIDBase, BLEUUID, BLEAdvData, BLEGapConnParams, NordicSemiException
-from pc_ble_driver_py.ble_driver      import ATT_MTU_DEFAULT
-from pc_ble_driver_py.ble_adapter     import BLEAdapter, BLEAdapterObserver, EvtSync
+HOST = ''   # Symbolic name meaning all available interfaces
+PORT =  int(args.port)
 
-global nrf_sd_ble_api_ver
+# global nrf_sd_ble_api_ver
 nrf_sd_ble_api_ver = config.sd_api_ver_get()
-logger  = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
-class BLE_Serial(BLEDriverObserver, BLEAdapterObserver):
-    BASE_UUID   = BLEUUIDBase([0x6E, 0x40, 0x00, 0x00, 0xB5, 0xA3, 0xF3, 0x93,
+
+class BleSerial(BLEDriverObserver, BLEAdapterObserver):
+    BASE_UUID = BLEUUIDBase([0x6E, 0x40, 0x00, 0x00, 0xB5, 0xA3, 0xF3, 0x93,
                                0xE0, 0xA9, 0xE5, 0x0E, 0x24, 0xDC, 0xCA, 0x9E])
 
-    RX_UUID     = BLEUUID(0x0003, BASE_UUID)
-    TX_UUID     = BLEUUID(0x0002, BASE_UUID)
-    LOCAL_ATT_MTU     = 23
-    def __init__(self, comport, periph_name, s_conn):
-        driver           = BLEDriver(serial_port    = comport,
-                                     baud_rate      = 115200)
-        adapter          = BLEAdapter(driver)
-        self.evt_sync           = EvtSync(['connected', 'disconnected'])
+    RX_UUID = BLEUUID(0x0003, BASE_UUID)
+    TX_UUID = BLEUUID(0x0002, BASE_UUID)
+    LOCAL_ATT_MTU = 23
+
+    def __init__(self, com_port, periph_name, s_conn):
+        driver = BLEDriver(serial_port    = com_port,
+                           baud_rate      = 1000000)
+        adapter = BLEAdapter(driver)
+        self.evt_sync = EvtSync(['connected', 'disconnected'])
```

Comment:
These changes are also unrelated to conversion to python 3 as far as I can tell.

They're not fully PEP 8 compliant either -- you would say `serial_port=com_port` and `baud_rate=1000000` (with no extra whitespace) in PEP 8 style.

Reply:
whitespace removed